From 7b9942946f8a9d4d9e83cfb34ab62198071c74c0 Mon Sep 17 00:00:00 2001
From: "H. Vetinari" <h.vetinari@gmx.com>
Date: Fri, 23 Oct 2026 18:27:32 +1100
Subject: [PATCH 2/2] add `REQUIRES: asserts` to
 `inst-select-extendedLLTs{,-err}.mir` due to interaction with -DNDEBUG

Suggested-by: Shoreshen <372660931@qq.com>
---
 .../CodeGen/AMDGPU/GlobalISel/inst-select-extendedLLTs-err.mir   | 1 +
 llvm/test/CodeGen/AMDGPU/GlobalISel/inst-select-extendedLLTs.mir | 1 +
 2 files changed, 2 insertions(+)

diff --git a/llvm/test/CodeGen/AMDGPU/GlobalISel/inst-select-extendedLLTs-err.mir b/llvm/test/CodeGen/AMDGPU/GlobalISel/inst-select-extendedLLTs-err.mir
index f067ab577a7d..c2e886b538eb 100644
--- a/llvm/test/CodeGen/AMDGPU/GlobalISel/inst-select-extendedLLTs-err.mir
+++ b/llvm/test/CodeGen/AMDGPU/GlobalISel/inst-select-extendedLLTs-err.mir
@@ -1,4 +1,5 @@
 # RUN: llc -mtriple=amdgpu12.51 -run-pass=instruction-select -verify-machineinstrs -global-isel-abort=2 -pass-remarks-missed='gisel*' %s -o /dev/null 2>&1 | FileCheck -check-prefix=ERR %s
+# REQUIRES: asserts
 
 # ERR: remark: <unknown>:0:0: cannot select: %2:vcc(s1) = G_FCMP floatpred(oeq), %1:vgpr(s32), %0:vgpr (in function: g_fcmp_op2_fconstant_s32)
 
diff --git a/llvm/test/CodeGen/AMDGPU/GlobalISel/inst-select-extendedLLTs.mir b/llvm/test/CodeGen/AMDGPU/GlobalISel/inst-select-extendedLLTs.mir
index 68346d08c6b2..69cfc291424d 100644
--- a/llvm/test/CodeGen/AMDGPU/GlobalISel/inst-select-extendedLLTs.mir
+++ b/llvm/test/CodeGen/AMDGPU/GlobalISel/inst-select-extendedLLTs.mir
@@ -1,5 +1,6 @@
 # NOTE: Assertions have been autogenerated by utils/update_mir_test_checks.py
 # RUN: llc -mtriple=amdgpu12.51 -run-pass=instruction-select -verify-machineinstrs -global-isel-abort=2 -pass-remarks-missed='gisel*' %s -o - 2> %t | FileCheck -check-prefix=GCN %s
+# REQUIRES: asserts
 
 # G_ADD i32: -> tablegen S_ADD_I32;
 # G_ADD s32: -> manual S_ADD_U32; selectG_ADD_SUB
